flake.nix

```diff
@@ -63,6 +64,27 @@
           };
 
           defaultPackage = flake.packages.dateilager;
+
+          devShell = pkgs.mkShell {
+            buildInputs = with pkgs; [
+              flake.packages.go
+              flake.packages.nodejs
+              flake.packages.postgresql
+              flake.packages.dev
+              flake.packages.clean
+              git
+              protobuf
+              protoc-gen-go
+              protoc-gen-go-grpc
+              go-migrate
+              mkcert
+            ];
```

Comment:
This makes the devShell build all the stuff you need to work on DL into your shell, but doesn't build DL itself. I always found that annoying with the original setup that I think Kira created that cd-ing into the dir built the whole project, which doesn't work so good when you are working on it and it might be busted.

Reply:
I'm down to properly Nix convert the project, it was in a very half converted state for a while.

But I do really miss the ability to control package versions, if ever Nix updates a package before we want to we're SOL.

Reply:
It's possible in nix, it's just annoying! You add a new dep at the top of a file for a nixpkgs sha that has the version of the thing you actually want, and then you refer to that version of nixpkgs in your exported packages! We did it in gadget a couple times when we didn't want to upgrade things.